main.go

```diff
@@ -25,22 +26,32 @@ import (
 
 func main() {
 	var (
-		lokiUrl   string
-		username  string
-		password  string
-		rawLabels string
+		lokiUrl              string
+		username             string
+		password             string
+		rawLabels            string
+		interruptWaitSeconds int64
 	)
 
 	fs := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
 	fs.StringVar(&lokiUrl, "addr", "", "Server address. Defaults to $LOKI_ADDR if not set.")
 	fs.StringVar(&username, "username", "", "Username for basic auth. Defaults to $LOKI_USERNAME if not set.")
 	fs.StringVar(&password, "password", "", "Password for basic auth. Defaults to $LOKI_PASSWORD if not set.")
 	fs.StringVar(&rawLabels, "labels", `{job="lokitee"}`, `Labels to inject for logs. i.e., {app="shell"}`)
+	fs.Int64Var(&interruptWaitSeconds, "interrupt-wait", 0, "Number of seconds to delay exiting if sending an interrupt signal like SIGTERM is sent")
 
 	if err := fs.Parse(os.Args[1:]); err != nil {
 		abort("error: could not parse flags: %s", err)
 	}
 
+	go func() {
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		sig := <-c
+		time.Sleep(time.Duration(interruptWaitSeconds) * time.Second)
+		abort("receiving signal '%s'", sig.String())
```

Comment:
nit: this would exit with os.Exit(1), which might not be correct for an interrupt. Maybe just silently `os.Exit(0)` here after the sleep?